src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -378,15 +535,16 @@ public void acceptRecord(final SAMRecord rec) {
 
             // Now do the bait overlaps
             int mappedBases = 0;
-            for (final AlignmentBlock block : rec.getAlignmentBlocks()) mappedBases += block.getLength();
+            for (final AlignmentBlock block : rec.getAlignmentBlocks()) mappedBases += getBasesAtMinimumQualityInBlock(rec, block);
             int onBaitBases = 0;
 
             if (probes != null && !probes.isEmpty()) {
                 for (final Interval bait : probes) {
                     for (final AlignmentBlock block : rec.getAlignmentBlocks()) {
                         final int end = CoordMath.getEnd(block.getReferenceStart(), block.getLength());
-
-                        for (int pos=block.getReferenceStart(); pos<=end; ++pos) {
+                        int readPos = block.getReadStart(); // one-based
+                        for (int pos=block.getReferenceStart(); pos<=end; ++pos, ++readPos) {
+                            if (baseQualities[readPos-1] < minimumBaseQuality) continue; // ignore low-quality bases
```

Comment:
Why not just add this as a third condition to the if() directly below instead of a separate if() and continue?